extensions/notification/configManager/MongoConfigManager.js

```diff
@@ -0,0 +1,291 @@
+const joi = require('joi');
+const semver = require('semver');
+
+const { ZenkoMetrics } = require('arsenal').metrics;
+const LRUCache = require('arsenal').algorithms
+    .cache.LRUCache;
+const MongoClient = require('mongodb').MongoClient;
+const ChangeStream = require('../../../lib/wrappers/ChangeStream');
+const { constructConnectionString, getMongoVersion } = require('../../utils/MongoUtils');
+
+const paramsJoi = joi.object({
+    mongoConfig: joi.object().required(),
+    bucketMetastore: joi.string().required(),
+    logger: joi.object().required(),
+}).required();
+
+const MAX_CACHED_ENTRIES = Number(process.env.MAX_CACHED_BUCKET_NOTIFICATION_CONFIGS)
+    || 1000;
+
+// should equal true if config manager's cache was hit during a get operation
+const CONFIG_MANAGER_CACHE_HIT = 'cache_hit';
+// Type of operation performed on the cache
+const CONFIG_MANAGER_OPERATION_TYPE = 'op';
+
+const cacheUpdates = ZenkoMetrics.createCounter({
+    name: 's3_notification_config_manager_cache_updates_total',
+    help: 'Total number of cache updates',
+    labelNames: [
+        CONFIG_MANAGER_OPERATION_TYPE,
+    ],
+});
+
+const configGetLag = ZenkoMetrics.createHistogram({
+    name: 's3_notification_config_manager_config_get_seconds',
+    help: 'Time it takes in seconds to get a bucket notification config from MongoDB',
+    labelNames: [
+        CONFIG_MANAGER_CACHE_HIT,
+    ],
+    buckets: [0.001, 0.01, 1, 10, 100, 1000],
+});
+
+const cachedBuckets = ZenkoMetrics.createGauge({
+    name: 's3_notification_config_manager_cached_buckets_count',
+    help: 'Total number of cached buckets in the notification config manager',
+});
+
+function onConfigManagerCacheUpdate(op) {
+    cacheUpdates.inc({
+        [CONFIG_MANAGER_OPERATION_TYPE]: op,
```

Comment:
```suggestion
        CONFIG_MANAGER_OPERATION_TYPE: op,
```

Reply:
We have to use the brackets else the label literally becomes `CONFIG_MANAGER_OPERATION_TYPE`